LEAF_Nexus/sources/Data.php

```diff
@@ -431,6 +439,10 @@ public function getAllTags($uid)
 
     public function addTag($uid, $tag)
     {
+        if(!is_numeric($uid)) {
```

Comment:
The assertion of `!is_numeric($UID)` should be removed since there is no guarantee that all classes that inherit from `Data.php` will have numeric UIDs.